### PR TITLE
[NTOS] Fancy blue screen with elegant stack backtrace

### PR DESCRIPTION
**Pull request by feel-the-dz3n**
[_This is the continuation of my previous BSoD-related pull request._ ](https://github.com/reactos/reactos/pull/1171)

## Purpose

BugCheck screen is the main sight of ReactOS. In my previous PR, I made screen look like in Windows 10, it even included QR-code library. But I realized - it was a bad idea. Reasons:
 - VGA screen resolution (640x480x4)
 - No place for QR-code, so It will be unreadable
 - Image is stretched on modern monitors, especially on widescreen ones

So I decided to close that Pull Request and start working on another one. And here it is. This is more user-friendly blue screen, it includes a lot of useful information for developers, and there is no sad emoji. 😞 

image
image
image

JIRA issue: [CORE-15474](https://jira.reactos.org/browse/CORE-15474)

## Proposed changes

- Added bitmap instead of intro texts. It has special palette for text and background. 
  _Remark:_ Have you seen Mizu ReactOS build? It's designed by @foxlet and was shared in some chats. Here how it looks:
image
Bitmap palette uses Mizu color scheme, not Windows 8 one.
- All bugcodes texts were removed/trimmed. I don't even know is it a good idea, but Windows 8+ even not uses any explanations for bug codes.
- Removed intro texts on screen.
- Added nice stack backtrace text on screen. 

## TODO

- [ ] Review bugcodes
- [ ] Test BugCheck behaviour. 
   I have already tested:
   - Ctrl+Double Scroll Lock
   - Killing winlogon.exe
   - Removing important Win32 components
   - INACCESSIBLE BOOT DEVICE on real hardware (booting from USB) [(video)](https://youtu.be/K2Pv9ijmZjw)

   In these situations it looked nice.

## I wanna answer before you ask/complain

Q: Intro text has different font and style unlike other text
A: Why not?

Q: Stack backtrace shows not all entries
A: It shows only as much as possible on screen to not lose important information above (bugcheck code, stop codes, etc.)

Q: Make switching between blue screen styles using registry/file/holding Shift/little pony
A: No, blue screen is the "ending station" of the system. At this moment OS has no access to things, especially to files.

_I finished._

### CODE REVIEW

**Comment by feel-the-dz3n** [comment]
Trying on real hardware. 

[Preview](http://www.youtube.com/watch?v=K2Pv9ijmZjw "ReactOS - New BSoD")

Video: https://youtu.be/K2Pv9ijmZjw

File: ntoskrnl/ke/bug.c [hunk 1]
```diff
@@ -643,28 +645,138 @@ KiDumpParameterImages(IN PCHAR Message,
...
 
+        RtlStringCbPrintfA(Buffer,BufferSize, "%2d %-8p %12s:%-8p", Iteration, KiStackBacktrace[Iteration].Frame, AnsiName, KiStackBacktrace[Iteration].FramePc);
+    }
+    else
+        RtlStringCbPrintfA(Buffer, BufferSize, "%2d %-8p %21p", Iteration, KiStackBacktrace[Iteration].Frame, KiStackBacktrace[Iteration].FramePc);
+}
+
+VOID
+NTAPI
+KiDoStackBacktrace()
```
Comment (learn-more) [inline ntoskrnl/ke/bug.c]: This function does not handle garbage or bad frames well.
Why not use `RtlCaptureStackBackTrace` ?

**Comment by tkreuzer** [comment]
If I understand the output correctly, the last row is supposed to show the last valid entry in the backtrace. I find it quite confusing though. I suggest instead to remove any NULL entries as seen in the 3rd screenshot and in case of more entries than displayable, just add a last line with something like "and 7 more"

**Comment by feel-the-dz3n** [comment]
I have no time to complete that PR so I'll finish it later

**Comment by ghostslayer989** [comment]
Ah yes windows 10 bluescreen style. But still, it looks smooth and "elegant".

**Comment by feel-the-dz3n** [comment]
I have no time and no idea how to fix it

**Pull request closed without merging**